runtimevar/runtimeconfigurator/runtimeconfigurator.go

```diff
@@ -68,6 +80,98 @@ func Dial(ctx context.Context, ts gcp.TokenSource) (pb.RuntimeConfigManagerClien
+	runtimevar.DefaultURLMux().RegisterVariable(Scheme, new(lazyCredsOpener))
+}
+
+// lazyCredsOpener obtains Application Default Credentials on the first call
+// to OpenVariableURL.
+type lazyCredsOpener struct {
+	init   sync.Once
+	opener *URLOpener
+	err    error
+}
+
+func (o *lazyCredsOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	o.init.Do(func() {
+		creds, err := gcp.DefaultCredentials(ctx)
+		if err != nil {
+			o.err = err
+			return
+		}
+		client, _, err := Dial(ctx, creds.TokenSource)
+		if err != nil {
+			o.err = err
+			return
+		}
+		o.opener = &URLOpener{Client: client}
+	})
+	if o.err != nil {
+		return nil, fmt.Errorf("open variable %q: %v", u, o.err)
+	}
+	return o.opener.OpenVariableURL(ctx, u)
+}
+
+// Scheme is the URL scheme runtimeconfigurator registers its URLOpener under on runtimevar.DefaultMux.
+const Scheme = "runtimeconfigurator"
+
+// URLOpener opens runtimeconfigurator URLs like "runtimeconfigurator://myproject/mycfg/myvar",
+// where:
+//
+//   - The URL's host holds the GCP projectID.
+//   - The first element of the URL's path holds the GCP RuntimeConfigurator ConfigID.
+//   - The second element of the URL's path holds the GCP RuntimeConfigurator Variable Name.
+// See https://cloud.google.com/deployment-manager/runtime-configurator/
+// for more details.
+//
+// This opener supports the following query parameters:
+//
+//   - decoder: The decoder to use. Defaults to URLOpener.Decoder, or
+//       runtimevar.BytesDecoder if URLOpener.Decoder is nil.
+//       See runtimevar.DecoderByName for supported values.
+//   - wait: The poll interval; supported values are from time.ParseDuration.
+//       Defaults to 30s.
+type URLOpener struct {
+	// Client must be set to a non-nil client authenticated with
+	// Cloud RuntimeConfigurator scope or equivalent.
+	Client pb.RuntimeConfigManagerClient
+
+	// Decoder and Options can be specified at URLOpener construction time,
+	// or provided/overridden via URL parameters.
+	Decoder *runtimevar.Decoder
+	Options Options
+}
+
+// OpenVariableURL opens a runtimeconfigurator Variable for u.
+func (o *URLOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	q := u.Query()
+	if decoderName := q.Get("decoder"); decoderName != "" || o.Decoder == nil {
+		var err error
+		o.Decoder, err = runtimevar.DecoderByName(q.Get("decoder"))
+		if err != nil {
+			return nil, fmt.Errorf("open variable %q: invalid \"decoder\": %v", u, err)
+		}
+		q.Del("decoder")
+	}
+
+	if wait := q.Get("wait"); wait != "" {
+		var err error
+		o.Options.WaitDuration, err = time.ParseDuration(wait)
+		if err != nil {
+			return nil, fmt.Errorf("open variable %q: invalid \"wait\": %v", u, err)
+		}
+		q.Del("wait")
+	}
+
+	for param := range q {
+		return nil, fmt.Errorf("open variable %q: invalid query parameter %q", u, param)
+	}
+	var rn ResourceName
+	rn.ProjectID = u.Host
+	rn.Config, rn.Variable = path.Split(u.Path)
```

Comment:
Should we check whether these are properly set?

Reply:
Yes, done. `path.Split` is also the wrong thing here as it removes double `//` and things like that, so I changed to use `strings.Split` and added more tests.